src/expr/src/relation/mod.rs

```diff
@@ -2748,26 +2748,29 @@ impl RowSetFinishing {
 
         let limit = self.limit.unwrap_or(usize::MAX);
 
-        // The code below is logically equivalent to:
-        //
-        // let mut total = 0;
-        // for (_, count) in &rows[offset_nth_row..] {
-        //     total += count.get();
-        // }
-        // let return_size = std::cmp::min(total, limit);
-        //
-        // but it breaks early if the limit is reached, instead of scanning the entire code.
-        let return_size = rows[offset_nth_row..]
-            .iter()
-            .try_fold(0, |sum, (_, count)| {
-                let new_sum = sum + count.get();
-                if new_sum > limit {
-                    None
-                } else {
-                    Some(new_sum)
-                }
-            })
-            .unwrap_or(limit);
+        // Count how many rows we'd expand into, returning early from the whole function
+        // if we don't have enough memory to expand the result, or break early from the
+        // iteration once we pass our limit.
+        let mut num_rows = 0;
+        let mut num_bytes = 0;
+        for (row, count) in &rows[offset_nth_row..] {
+            num_rows += count.get();
+            num_bytes += row.byte_len();
```

Comment:
Ahh, so `row.byte_len()`is just a size of the single row, so we'd need to multiply it by `count.get()` because that's how many rows we'd end up creating?

Reply:
yes, exactly